torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -309,8 +310,11 @@ bool shouldAllCommunicatorsRegisterAllTensors() {
 // - This map has also to be maintained as global variable since the register
 //   hooks are called outside the scope of any PG, thus we need traverse
 //   communicators in all PGs.
-static std::unordered_map<std::shared_ptr<NCCLComm>, int> ncclCommDevIdxMap;
-static std::mutex ncclCommDevIdxMapMutex;
+using MemPoolSet = std::
+    unordered_set<c10::cuda::MempoolId_t, c10::hash<c10::cuda::MempoolId_t>>;
+static std::unordered_map<std::shared_ptr<NCCLComm>, MemPoolSet>
+    ncclCommMemPoolMap;
+static std::mutex ncclCommMemPoolMapMutex;
```

Comment:
nit: can we add comments to these variables?
We can leave them as global today, but later we will need to consider `thread_local`'ness, because ProcessGroupNCCL (in cpp) is supposed to support one GPU per thread.

Reply:
I updated the existing comments above